core/lib/types/src/kzg.rs

```diff
@@ -0,0 +1,315 @@
+use std::convert::TryInto;
+
+use c_kzg::{Blob, Bytes32, Bytes48, KzgCommitment, KzgProof, KzgSettings, BYTES_PER_BLOB};
+use zk_evm::{
+    sha2::Sha256,
+    sha3::{Digest, Keccak256},
+};
+use zkevm_circuits::eip_4844::{
+    input::{BLOB_CHUNK_SIZE, ELEMENTS_PER_4844_BLOCK},
+    zksync_pubdata_into_ethereum_4844_data,
+};
+
+const BYTES_PER_BLOB_ZK_SYNC: usize = BLOB_CHUNK_SIZE * ELEMENTS_PER_4844_BLOCK;
+
+/// Packed pubdata commitments.
+/// Format: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || opening proof (48 bytes)) = 144 bytes
+const BYTES_PER_PUBDATA_COMMITMENT: usize = 144;
+
+const VERSIONED_HASH_VERSION_KZG: u8 = 0x01;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct KzgInfo {
+    pub blob: Blob,
+    pub kzg_commitment: Bytes48,
+    pub opening_point: Bytes32,
+    pub opening_value: Bytes32,
+    pub opening_proof: Bytes48,
```

Comment:
please add a small comment on what does each field represent (esp. with regard to what is the difference between opening proof and blob proof)